src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+import java.util.Date;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+import java.text.SimpleDateFormat;
+
+
+public class CommentController {
+
+    public static Route saveComments = ((request, response) -> {
+        JacksonMongoCollection<CommentModel> mongoCollection =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        CommentModel commentModel;
+        SessionManager sessionManager = new SparkSessionManager(request);
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        String bodyText = paramHelper.valueOf("bodyText");
+        String authorId = sessionManager.getCurrentUserId();
+        String postId = paramHelper.valueOf("postId");
+        if((bodyText.isEmpty() || bodyText == null) && (authorId.isEmpty() || authorId == null) && (postId.isEmpty() || postId == null)){
+            return "bad";
+        }
+        commentModel = new CommentModel();
+        commentModel.setPostId(postId);
+        commentModel.setAuthorId(authorId);
+        commentModel.setBodyText(bodyText);
+        commentModel.setDate();
+        mongoCollection.save(commentModel);
+        return postId;
+    });
+
+    public static Route getComments = ((request, response) -> {
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        String postId = paramHelper.valueOf("postId");
+        String startAfterId = paramHelper.valueOf("startAfterId");
+        String date = paramHelper.valueOf("date");
+        PostModel post = mongoCollectionPost.findOne(eq("_id", new ObjectId(postId)));
+        List<CommentModel> commentModelList;
+        if(post == null){
+            return "post DNE";
+        }
+        if(startAfterId == null){
+            if(date == null){
+                Date thedate = new java.util.Date(System.currentTimeMillis());
+                DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+                date = dateFormat.format(thedate);
+            }
+            commentModelList = getByDate(postId, date);
+        }else{
+            commentModelList = getAfterId(postId, startAfterId);
+        }
+        if(postId == null){
+            return "not ok";
+        }
+        CommentListingModel commentList = new CommentListingModel();
+        commentList.count = commentModelList.size();
+        if (commentModelList.size() > 0){
+            commentList.lastId = commentModelList.get(commentModelList.size()-1)._id.toString();
+            commentList.postId = commentModelList.get(0).getPostIdString();
+        }
+        commentList.comments = commentModelList;
+        return commentList;
+
+    });
+
```

Comment:
### `GET /comments/list`
- `startAfterId` either broken or does not work as intended
- `limit` parameter is missing from endpoint
- `date` parameter not required from spec, but:
  - current implementation will only list comments from the current DAY when `startAfterId` is not supplied, not all comments for the specific post
- blank postId crashes server with 500 error, along with a blank request
  - should halt with an error instead